VultisigApp/VultisigApp/Views/Vault/CustomTokenView.swift

```diff
@@ -0,0 +1,257 @@
+                    Separator()
+                    getDetailsCell(for: "Decimals", with: tokenDecimals.description)
+                    Separator()
+                    
+                    if let logo = token?.logo, !logo.isEmpty {
+                        getDetailsCell(for: "Logo", with: logo)
+                        Separator()
+                    }
+                    if let priceRate = token?.priceRate, priceRate > 0 {
+                        getDetailsCell(for: "Price Rate", with: "\(priceRate)")
+                        Separator()
+                    }
+                    if let provider = token?.priceProviderId, !provider.isEmpty {
+                        getDetailsCell(for: "Price Provider ID", with: provider)
+                        Separator()
+                    }
+                    if let rawBalance = token?.rawBalance, !rawBalance.isEmpty {
+                        getDetailsCell(for: "Raw Balance", with: rawBalance)
+                        Separator()
+                    }
+                    getDetailsCell(for: "Is Native Token", with: "\(token?.isNativeToken ?? false)")
+                }
+                .padding(16)
+                .background(Color.blue600)
+                .cornerRadius(10)
+                
+                Button(action: {
+                    saveAssets()
+                }) {
+                    FilledButton(title: "Add \(tokenSymbol) token")
+                }
+            }
+            
+        }
+    }
+    
+    private func getAddressCell(for title: String, with address: String) -> some View {
+        VStack(alignment: .leading, spacing: 8) {
+            Text(NSLocalizedString(title, comment: ""))
+                .font(.body20MontserratSemiBold)
+                .foregroundColor(.neutral0)
+            
+            Text(address)
+                .font(.body12Menlo)
+                .foregroundColor(.turquoise600)
+        }
+        .frame(maxWidth: .infinity, alignment: .leading)
+    }
+    
+    private func getDetailsCell(for title: String, with value: String) -> some View {
+        HStack {
+            Text(NSLocalizedString(title, comment: ""))
+            Spacer()
+            Text(value)
+        }
+        .font(.body16MenloBold)
+        .foregroundColor(.neutral100)
+    }
+    
+    func errorView(error: Error) -> some View {
+        return VStack(spacing: 16) {
+            Text(error.localizedDescription)
+                .multilineTextAlignment(.center)
+                .font(.body16Menlo)
+                .foregroundColor(.neutral0)
+                .padding(.horizontal, 16)
+            
+            Button {
+                Task { await fetchTokenInfo() }
+            } label: {
+                FilledButton(title: "Retry")
+            }
+            .padding(.horizontal, 40)
+        }
+        .frame(maxWidth: .infinity, maxHeight: .infinity)
+    }
+    
+    private func fetchTokenInfo() async {
```

Comment:
we can just mark it as MainActor and avoid DispatchQueue.main.async.

Reply:
Fixed